File: backend/app/schemas/analysis.py
from pydantic import BaseModel


class SortableField(BaseModel):
    id: str
    label: str

    sortable: bool = True

    unique_values: int = 0

    total_labels: int = 0


class FieldStatistics(BaseModel):
    values: dict[str, int]


class AnalysisResult(BaseModel):
    marketplace: str

    page_count: int
    label_count: int

    sortable_fields: list[SortableField]

    courier_priority_options: list[str]
    field_statistics: dict[str, FieldStatistics]

File: frontend/src/types/analysis.ts
export interface SortableField {
  id: string;
  label: string;
  sortable: boolean;
  unique_values: number;
  total_labels: number;
}

export interface FieldStatisticsItem {
  values: Record<string, number>;
}

export interface AnalysisResult {
  marketplace: string;
  page_count: number;
  label_count: number;
  sortable_fields: SortableField[];
  courier_priority_options: string[];
  field_statistics: Record<string, FieldStatisticsItem>;
}
